packages/docs/src/routes/docs/deployments/self-hosting/index.mdx

```diff
@@ -0,0 +1,536 @@
+}
+```
+
+### Apache Configuration
+
+For Apache with mod_proxy enabled:
+
+```apache
+<VirtualHost *:443>
+    ServerName your-domain.com
+    
+    # SSL Configuration
+    SSLEngine on
+    SSLCertificateFile /path/to/cert.pem
+    SSLCertificateKeyFile /path/to/key.pem
+    
+    # Proxy to Node.js server (for Express/Fastify adapters)
+    ProxyPreserveHost On
+    ProxyPass / http://localhost:3000/
+    ProxyPassReverse / http://localhost:3000/
+    
+    # Serve static assets directly
+    <Directory /path/to/your/app/dist/build>
+        Options Indexes FollowSymLinks
+        AllowOverride All
+        Require all granted
+        
+        # Cache settings
+        <IfModule mod_expires.c>
+            ExpiresActive On
+            ExpiresDefault "access plus 1 year"
+            Header append Cache-Control "public, immutable"
+        </IfModule>
+    </Directory>
+    
+    Alias /build/ /path/to/your/app/dist/build/
+    
+    # Security headers
+    Header always set X-Content-Type-Options "nosniff"
+    Header always set X-Frame-Options "SAMEORIGIN"
+    Header always set X-XSS-Protection "1; mode=block"
+</VirtualHost>
+```
+
+## Important Security Considerations
+
+### CSRF Protection
+
+Qwik City applications are protected against CSRF attacks by default for all POST, PATCH, and DELETE form submissions. The `ORIGIN` environment variable is crucial for this protection.
+
+If you need to disable CSRF protection (not recommended for production), you can set `checkOrigin: false` in the adapter configuration:
+
+```tsx
+// src/entry.express.tsx or src/entry.fastify.tsx
+const { router, notFound } = createQwikCity({
+  render,
+  qwikCityPlan,
+  manifest,
+  checkOrigin: false, // Disable CSRF protection (not recommended)
+});
+```
+
+### Other Security Best Practices
+
+1. Always run your Node.js application as a non-root user
+2. Set up proper firewall rules to only expose necessary ports
+3. Use HTTPS with proper certificates (Let's Encrypt works well)
+4. Implement rate limiting for API endpoints
+5. Keep all dependencies updated regularly
+6. Configure Content-Security-Policy headers appropriate for your application
+7. Consider using a Web Application Firewall (WAF) for additional protection
+
+## Monitoring and Logs
+
+1. If using PM2:
+   - Monitor application: `pm2 monit`
+   - Check logs: `pm2 logs qwik-app`
+   - Set up PM2 dashboard: `pm2 plus`
+
+2. Logging options:
+   - Configure Nginx logs in `/var/log/nginx/`
+   - Use logging services like Winston or Pino for structured logs
+
+3. Consider adding application monitoring with:
+   - Prometheus + Grafana for self-hosted monitoring
+   - New Relic, Datadog, or Sentry for managed monitoring
+
+## Optimizing Performance
+
+Qwik's resumable architecture already provides excellent performance, but you can further optimize:
+
+1. Enable HTTP/2 or HTTP/3 in your web server
+2. Use a CDN for global edge caching
+3. Enable Brotli compression for smaller payload sizes
```

Comment:
It would be good to expand upon this, brotli is slow to compress so normally it is done on asset files before serving. So then the build/ and assets/ files should be pre-compressed with brotli and gzip without deleting the originals, and the web servers should be configured to use the compressed files.